### PR TITLE
Remove deploy step

### PR DESCRIPTION
**Pull request by rasolca**
Removing deploy step:
hard to maintain
doesn't reuse layers already in jfrog

Note: cannot use the MPS server available on daint and need to start it manually, otherwise tests hangs.
I suspect it is due to some file incompatibilities as this problem didn't show up with the "old" deployed images.

### CODE REVIEW

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

File: ci/cpu/asan_ubsan_lsan.yml [hunk 1]
```diff
@@ -17,9 +17,7 @@ cpu asan ubsan lsan build:
   needs:
     - cpu asan ubsan lsan deps
   variables:
-    DEPLOY_IMAGE: $CSCS_REGISTRY_PATH/cpu-asan-ubsan-lsan/deploy:$CI_COMMIT_SHA
-    # For symbolizing stacktraces with llvm-symbolizer
-    EXTRA_APTGET_DEPLOY: "llvm-18"
```
Comment (msimberg) [inline ci/cpu/asan_ubsan_lsan.yml:22]: I suppose `llvm-18` is pulled by `clang-18` and no longer needed explicitly?

File: ci/ctest_to_gitlab.sh [hunk 2]
```diff
@@ -30,7 +31,7 @@ variables:
   SLURM_EXCLUSIVE: ''
   SLURM_EXACT: ''
   SLURM_CONSTRAINT: $SLURM_CONSTRAINT
-  CRAY_CUDA_MPS: 1
+  CRAY_CUDA_MPS: 0
```
Comment (msimberg) [inline ci/ctest_to_gitlab.sh:34]: Can this be outright removed? Or do you prefer to keep it explicit since it can't be enabled?

**Review by msimberg — APPROVED** [review]

File: ci/ctest_to_gitlab.sh (hunk 2 above)
Comment (rasolca) [reply ci/ctest_to_gitlab.sh:34]: Want to keep to re-enable on alps.

File: ci/cpu/asan_ubsan_lsan.yml (hunk 1 above)
Comment (rasolca) [reply ci/cpu/asan_ubsan_lsan.yml:22]: We are not dealing with a clean image anymore. All the apt packages are installed in the `build deps` phase and clang-18 depends on llvm-18.

**Pull request merged**